crates/autopilot/src/decoded_settlement.rs

```diff
@@ -412,20 +375,23 @@ impl DecodedSettlement {
         tracing::trace!(?fee, "fee after conversion to native token");
 
         Some(Fees {
-            order: order.order_uid,
-            sell: solver_fee,
+            order,
+            sell: match order_fee {
+                Some(_) => ExecutedFee::Order(solver_fee),
+                None => ExecutedFee::Surplus(solver_fee),
+            },
             native: big_rational_to_u256(&fee).ok()?,
         })
     }
 }
 
-/// Computed executed fees for an order with solver-computed fees. These are
-/// computed based on-chain settlement data.
+/// These are computed based on-chain settlement data.
+#[derive(Debug)]
 pub struct Fees {
     /// The UID of the order associated with these fees.
     pub order: OrderUid,
     /// The executed fees in the sell token.
-    pub sell: U256,
+    pub sell: ExecutedFee,
```

Comment:
Conceptually, why do we need to differentiate here between Surplus and Order fee subtypes? Isn't the amount enough?

Specifically, I don't see where we read the `solver_fee` column from the database. I recall this is for solver rewards, but aren't we using the total fee on the batch for that?

Reply:
I am still not sure if we definitely want to merge `order_execution::solver_fee` and `order_execution::surplus_fee` into a single column, although that was my main driver for the refactor from the beggining.
When we switch to all orders are limit orders and use quote to determine the order class, it becomes possible to submit a limit order with `fee_amount != 0`. 
I asked this on slack.

Reply:
I might be overthinking here. Reverted this change and now I am saving only surplus_fee for limit orders to the `order_execution::surplus_fee` field. As you said, we no longer use `order_execution::solver_fee` anywhere.

Reply:
Resolved.